observer.py

```diff
@@ -112,32 +114,58 @@ async def run_alerts():
                             coingecko_price=coingecko_price,
                             include_noisy=args.include_noisy_alerts,
                         )
+                        price_errors = validators[symbol].verify_price(
+                            price=price,
+                            include_noisy=args.include_noisy_alerts,
+                        )
+
                         if price_account_errors:
                             errors.extend(price_account_errors)
 
+                        if price_errors:
+                            errors.extend(price_errors)
+
                         for price_comp in price_account.price_components:
                             # The PythPublisherKey
-                            publisher = price_comp.publisher_key.key
+                            publisher_key = price_comp.publisher_key.key
 
-                            price.quoters[publisher] = price_comp.latest_price_info
-                            price.quoter_aggregates[
-                                publisher
-                            ] = price_comp.last_aggregate_price_info
+                            price.quoters[publisher_key] = price_comp.latest_price_info
+                            price.quoter_aggregates[publisher_key] = price_comp.last_aggregate_price_info
 
                             if args.enable_prometheus:
-                                gprice.labels(
+                                prom_prices.labels(
                                     symbol=symbol,
-                                    publisher=publisher,
-                                    status=price.quoters[publisher].price_status.name,
-                                ).set(price.quoters[publisher].price)
+                                    publisher=publisher_key,
+                                    status=price.quoters[publisher_key].price_status.name,
+                                ).set(price.quoters[publisher_key].price)
+
+                                # Write OK as empty error code
+                                if not price_errors and not price_account_errors:
+                                    prom_publisher_status.labels(
+                                        symbol=symbol,
+                                        publisher=publisher_key,
```

Comment:
I think the way the errors work is that they have a publisher_key field that can be None. If the key is None, it's an error for the symbol as a whole, not a particular publisher. I think setting publisher=publisher_key here is not the right thing to do.

I also suggest moving all of the error logging code outside the for loop, so it's adjacent to the `notify` call below. That's where we are currently logging errors to slack.

Reply:
You might want to make another Gauge for errors related to the symbol and not a particular publisher.

Reply:
Got it. One caveat is that we might still want to report an "ok" status for each publisher so we be sure that the lack of errors means everything is running smoothly vs. the observer service is down and isn't reporting errors when it should.

Reply:
i see. I think this line doesn't correctly report the OK status for each publisher though, as an error on publisher A will result in an error report for publisher B. You will have to check for each publisher whether there are any errors with their publisher_key in the list.